### PR TITLE
Use SKStoreReviewController

### PR DESCRIPTION
**Pull request by stephencelis**
10.3 introduced [SKStoreReviewController](https://developer.apple.com/documentation/storekit/skstorereviewcontroller), which will be the required way to prompt users to leave a review. On the plus side, it allows users to rate inline without leaving the app!

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/ThanksViewController.swift [hunk 1]
```diff
@@ -185,18 +186,22 @@ internal final class ThanksViewController: UIViewController, UICollectionViewDel
...
-        }, remindHandler: { [weak self] _ in
-          self?.viewModel.inputs.rateRemindLaterButtonTapped()
-        }, noHandler: { [weak self] _ in
-          self?.viewModel.inputs.rateNoThanksButtonTapped()
-      }),
-      animated: true,
-      completion: nil
-    )
+    if #available(iOS 10.3, *) {
+      SKStoreReviewController.requestReview()
```
Comment (stephencelis) [inline Kickstarter-iOS/Views/Controllers/ThanksViewController.swift]: There's no input to feed back into our view model. There's no mechanism to know if the alert was displayed, if the user reviewed or canceled, etc.

Comment (theginbin) [inline Kickstarter-iOS/Views/Controllers/ThanksViewController.swift]: oh that's a bummer

**Comment by justinswart** [comment]
@theginbin I think we should, currently it just takes us to the app store.

**Comment by stephencelis** [comment]
> looks good! should we use this method in the SettingsViewController for sending people to the store?

Unfortunately we can't! Per Apple's guidelines this function shouldn't be called as a direct response to a user action. Apple determines whether or not to do anything at all when the function is called based on these requirements:

- Has the user disabled app review notifications in settings?
- Has the user seen an app review notification this version?

**Pull request merged**